File: solutions/Reverse-Integer/Reverse-Integer.py

class Solution:
    # @return an integer
    def reverse(self, x):
        flag=False
        if x==0:
            return x
        elif x<0:
            flag=True
            x=-x
        ans=0
        while x:
            ans=ans*10+x%10
            x/=10
        if flag:
            return -ans
        else:
            return ans        
 

File: solutions/Reverse-Integer/Reverse-Integer.swift
class Solution {
    func reverse(_ x: Int) -> Int {
        var n = x
        var ans = 0
        while n != 0 {
            ans = ans * 10 + n % 10
            if ans >= Int32.max || ans <= Int32.min {
                return 0
            }
            n /= 10
        }
        return ans
    }
}